packages/opentelemetry-core/src/context/propagation/B3Propagator.ts

```diff
@@ -34,20 +36,9 @@ export const PARENT_SPAN_ID_KEY = Context.createKey(
 export const DEBUG_FLAG_KEY = Context.createKey(
   'OpenTelemetry Context Key B3 Debug Flag'
 );
-const VALID_TRACEID_REGEX = /^([0-9a-f]{16}){1,2}$/i;
```

Comment:
B3 allows 16 or 32 character trace IDs. This changes the propagator to not be spec compliant. Please remove the changes to the B3 propagator.

Reply:
I don't think it does because on extract we pad the the traceid in this function `getTraceId`

Reply:
Ah I see the padding is done before this check. Good call

Reply:
Yeah, and on inject we already expect it to be 32 characters, right?

Reply:
If not, then I'll change it back.